libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -1573,8 +1574,7 @@ string YulUtilFunctions::copyArrayToStorageFunction(ArrayType const& _fromType,
 			if (_fromType.baseType()->isDynamicallyEncoded())
 				templ("accessCalldataTail", accessCalldataTailFunction(*_fromType.baseType()));
 		}
-		if (_toType.isDynamicallySized())
-			templ("resizeArray", resizeDynamicArrayFunction(_toType));
+		templ("resizeArray", resizeArrayFunction(_toType));
 		templ("arrayLength",arrayLengthFunction(_fromType));
```

Comment:
I'd add an assert that if both types are statically-sized arrays, `_toType` must not be shorter than `_fromType`.

Reply:
Added more assertion at the beginning of a function